dace/frontend/python/newast.py

```diff
@@ -3952,13 +3952,9 @@ def visit_Str(self, node: ast.Str):
         return node.s
 
     def visit_Num(self, node: ast.Num):
-        if isinstance(node.n, (int, float, complex, bool)):
```

Comment:
Without this change, python frontend programs containing scalars cannot be serialized, since numpy scalars are not serializable. There is a smaller regression test case here:

https://github.com/spcl/dace/pull/608/commits/d1e02087d2adcf4737ef12aa5e22492265df8c6c

Reply:
The issue here is that this is the mechanism through which the `default_types` flag is enforced. I am not sure (without checking an example) what happens. I guess everything becomes 64-bit. This can be a serious issue for FPGAs. I haven't looked at the whole PR, so I don't know if there is a countermeasure for this elsewhere.

Reply:
now fixed without removing those lines.